CORE-940 User can customize data columns

ash3rz opened the pull request:
For now, I've made it the default that only the checkbox, name, and dot menu show in the table view.  I removed the breakpoints because I figure users can set whatever columns make sense for their screen size as needed.  In another PR, I can probably update the default columns to check for a value stored in local storage so the setting persists.

Desktop view:
![image](https://user-images.githubusercontent.com/8909156/76805129-4137f700-679b-11ea-8a45-6f46497c422f.png)

Menu open:
![image](https://user-images.githubusercontent.com/8909156/77586908-2d2f7c00-6ea4-11ea-82b8-cdea906ec220.png)

Mobile view:
![image](https://user-images.githubusercontent.com/8909156/77586982-4b957780-6ea4-11ea-843a-5d942f721b1f.png)

Scrolling on mobile:
![customcolumns](https://user-images.githubusercontent.com/8909156/77586769-f48fa280-6ea3-11ea-8180-480de6b6ad1e.gif)

sri2k1us commented:
Code LGTM. I have few concerns / questions though:
- I am not sure about removing breakpoints. I agree users have the power to choose the columns that fits the screen size. What happens when users chose to view to more than what the screen can fit ? Will there be a horizontal scroll. A screenshot of this behavior will be good

- I agree with @psarando comment on using check mark for selection. Text shift looks odd.

ash3rz commented:
Thanks for the reviews!

I've updated the screenshots and added a gif to show that you can scroll horizontally.  I felt on the fence about getting rid of breakpoints too, so we can revisit if needed.  I started off leaving them, but then I wasn't sure what to do if a user, for example, added all the columns and then made their browser window smaller or if their browser already couldn't fit all the columns. It seemed confusing to start removing/hiding columns in that case especially if scroll works, so I removed the breakpoints.  I'm totally open to other ideas on this though.
